gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+        Dns.ZoneListOption.fields(Dns.ZoneField.DESCRIPTION)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNotNull(zone.description());
+    assertNull(zone.nameServerSet());
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.NAME_SERVERS)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNull(zone.description());
+    assertNull(zone.nameServerSet());
+    assertTrue(!zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.NAME_SERVER_SET)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNull(zone.description());
+    assertNull(zone.nameServerSet()); // we cannot set it using gcloud java
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    // several combined
+    zones = filter(DNS.listZones(Dns.ZoneListOption.fields(Dns.ZoneField.ZONE_ID,
+        Dns.ZoneField.DESCRIPTION),
+        Dns.ZoneListOption.pageSize(1)).iterateAll());
+    assertEquals(2, zones.size());
+    for (Zone current : zones) {
+      assertNull(current.creationTimeMillis());
+      assertNotNull(current.name());
+      assertNull(current.dnsName());
+      assertNotNull(current.description());
+      assertNull(current.nameServerSet());
+      assertTrue(zone.nameServers().isEmpty());
+      assertNotNull(current.id());
+    }
+  }
+
+  @Test
+  public void testDeleteZoneUsingServiceObject() {
+    Zone created = DNS.create(ZONE1);
+    assertEquals(created, DNS.getZone(ZONE1.name()));
+    DNS.delete(ZONE1.name());
+    assertNull(DNS.getZone(ZONE1.name()));
+  }
+
+  @Test
+  public void testDeleteZoneUsingZoneObject() {
+    Zone created = DNS.create(ZONE1);
+    assertEquals(created, DNS.getZone(ZONE1.name()));
+    created.delete();
+    assertNull(DNS.getZone(ZONE1.name()));
+  }
+
+  private void assertEqChangesIgnoreStatus(ChangeRequest expected, ChangeRequest actual) {
+    ChangeRequest unifiedEx = ChangeRequest.fromPb(expected.toPb().setStatus("pending"));
+    ChangeRequest unifiedAct = ChangeRequest.fromPb(actual.toPb().setStatus("pending"));
+    assertEquals(unifiedEx, unifiedAct);
+  }
+
+  private static void checkChangeComplete(String zoneName, String changeId) {
+    for (int i = 0; i < TIME_LIMIT; i++) {
```

Comment:
Just throwing some ideas. What happens if this iteration limit is exceeded? Shouldn't the integration test fail? 

I think elsewhere (e.g. Bigquery jobs or with storage bucket deletion) we loop until the condition is verified and we have a timeout for tests:

```java
@Rule
public Timeout globalTimeout = Timeout.seconds(300);
```
That makes the rest fail if taking too long.